clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/services/ContainerInformationService.java

```diff
@@ -138,13 +153,27 @@ public boolean taskHasHealthCheck(Service service, String accountName, String re
   }
 
   private String toPlatformHealthState(
-      String ecsTaskStatus, String ecsTaskHealthStatus, boolean hasHealthCheck) {
+      String ecsTaskStatus,
+      String ecsTaskHealthStatus,
+      boolean hasHealthCheck,
+      EcsTargetHealth ecsTargetHealth) {
     if (hasHealthCheck && "UNKNOWN".equals(ecsTaskHealthStatus)) {
       return "Starting";
     } else if ("UNHEALTHY".equals(ecsTaskHealthStatus)) {
       return "Down";
     }
 
+    if (ecsTargetHealth != null) {
+      List<String> statuses =
+          ecsTargetHealth.getTargetHealthDescriptions().stream()
+              .map(tg -> tg.getTargetHealth().getState())
+              .distinct()
+              .collect(Collectors.toList());
+      if (statuses.stream().anyMatch(it -> it.equalsIgnoreCase("unhealthy"))) {
```

Comment:
OK so the downside with this still, we're ONLY checking "unhealthy", and if that's NOT returned you fall back onto the ECS task status checks.  This leads to a situation where you could have:
* ECS Task Check:  HEALTHY
* ECS LB CHeck: PENDING/UNKNOWN - as it has NOT passed LB health checks which may be different.

This would put it back into service incorrectly in this case.  As the LB health check if set should trump the ECS task service health checks when configured.